sqlmesh/core/engine_adapter/spark.py

```diff
@@ -59,6 +60,22 @@ def insert_append(
         else:
             super().insert_append(table_name, query_or_df, columns_to_types, contains_json)
 
+    def merge(
+        self,
+        target_table: TableName,
+        source_table: QueryOrDF,
+        column_names: t.Iterable[str],
```

Comment:
we should change the types of column_names to seq or something, this is an iterable, and you iterate on it twice which is not correct